src/pj_datum_set.c

```diff
@@ -102,6 +102,8 @@ int pj_datum_set(projCtx ctx, paralist *pl, PJ *projdef)
 
         projdef->datum_type = PJD_GRIDSHIFT;
         projdef->catalog_name = strdup(catalog);
+        if (!projdef->catalog_name)
+            return 1;   /* TODO: Should we set errno? */
```

Comment:
Yes - I believe calling `pj_ctx_set_errno (ctx, ENOMEM)` would be the proper thing to do here.